internal/controller/csiaddons/networkfence_controller.go

```diff
@@ -59,6 +59,16 @@ func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
 	if nwFence == nil {
 		return errors.New("NetworkFence resource is empty")
 	}
+
+	if nwFence.Spec.NetworkFenceClassName != "" {
+		if nwFence.Spec.Cidrs == nil {
+			return errors.New("required parameter cidrs is not specified")
+		}
+
+		// Driver name and secrets will be read (and validated) in NetworkFenceClass
+		return nil
+	}
```

Comment:
why these 2 are tied to each other?

Reply:
If classname is specified, only CIDRs are validated. Provisioner and secrets are validated inside NFClass.